internal/system/agent/getexecutor.go

```diff
@@ -0,0 +1,134 @@
+		result.Configuration[service] = response.Process(config, ce.loggingClient)
+	}
+	return result
+}
+
+func (ce getExecutor) getConfig(service string, ctx context.Context) (string, error) {
+	var result string
+
+	// Check whether SMA does _not_ know of ServiceKey ("service") as being one for one of its ready-made list of clients.
+	if !IsKnownServiceKey(service) {
+		ce.loggingClient.Info(fmt.Sprintf("service %s not known to SMA as being in the ready-made list of clients", service))
+
+		// Service unknown to SMA, so ask the Registry whether `service` is available.
+		err := ce.registryClient.IsServiceAvailable(service)
+		if err != nil {
+			result = fmt.Sprintf(err.Error())
+			ce.loggingClient.Error(err.Error())
+		} else {
+			ce.loggingClient.Info(fmt.Sprintf("Registry responded with %s service available", service))
+
+			// Since service is unknown to SMA, ask the Registry for a ServiceEndpoint associated with `service`
+			e, err := ce.registryClient.GetServiceEndpoint(service)
+			if err != nil {
+				result = fmt.Sprintf("on attempting to get ServiceEndpoint for service %s, got error: %v", service, err.Error())
+				ce.loggingClient.Error(err.Error())
+			} else {
+				// Preparing to add the specified key to the map where the value will be the respective GeneralClient
+				clientInfo := config.ClientInfo{}
+				clientInfo.Protocol = ce.serviceProtocol
+				clientInfo.Host = e.Host
+				clientInfo.Port = e.Port
+
+				// This code will evolve to take into account a manifest-like functionality in future. So
+				// rather than assume that the runtime bool flag useRegistry has been initialized to true,
+				// given that the flow has reached this point, having already called functions on the Registry,
+				// such as RegistryClient.IsServiceAvailable(service), we test for its truthiness. I expect
+				// this code to be refactored as we evolve toward a manifest-like functionality in future.
+				usingRegistry := false
+				if ce.registryClient != nil {
+					usingRegistry = true
+				}
+
+				ce.configClients[e.ServiceId] = clientInfo
+				params := types.EndpointParams{
+					ServiceKey:  e.ServiceId,
+					Path:        "/",
+					UseRegistry: usingRegistry,
+					Url:         ce.configClients[e.ServiceId].Url() + clients.ApiConfigRoute,
+					Interval:    internal.ClientMonitorDefault,
+				}
+				responseJSON, err := general.NewGeneralClient(params, startup.Endpoint{RegistryClient: &ce.registryClient}).FetchConfiguration(ctx)
+				if err != nil {
+					result = fmt.Sprintf(err.Error())
+					ce.loggingClient.Error(err.Error())
+				} else {
+					result = responseJSON
+				}
+				return result, nil
+			}
+		}
+	} else {
+		// Service is known to SMA, so no need to ask the Registry for a ServiceEndpoint associated with `service`
+		// Simply use one of the ready-made list of clients.
+		// TODO: Revisit and fix. The following snippet was merely to address a merge conflict after the step to
```

Comment:
- Currently, the following statement creates a panic because `client` is `nil` in `client, ok = c.clients[clientName]`:
gcClient, b := ce.genClients.Get(service)
- Investigating how to get it properly initialized.

Reply:
Consider refactoring to remove the multitude of if/else statements.

A similar example can be found here: https://github.com/edgexfoundry/edgex-go/blob/16192152b3c66082da2350bd202f4259c7319125/internal/system/agent/direct/metrics.go#L66